content/courses/program-optimization/program-configuration.md

```diff
@@ -537,58 +547,100 @@ Next, update the `config.ts` test file to create a mint using the generated
 keypair. Start by deleting the `mint` constant.
 
 ```typescript
-const mint = new anchor.web3.PublicKey(
-  "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
-);
+const USDC_MINT = new PublicKey("EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v");
+```
+
+<Callout>
+
+The `anchor test` command, when run on a local network, starts a new test
+validator using `solana-test-validator`. This test validator uses a
+non-upgradeable loader. The non-upgradeable loader makes it so the program's
+`program_data` account isn't initialized when the validator starts. You'll
+recall from the lesson that this account is how we access the upgrade authority
+from the program. </Callout>
+
+To work around this, you can add a `deploy` function to the test file that runs
+the deploy command for the program with an upgradeable loader. To use it, run
+`anchor test --skip-deploy`, and call the `deploy` function within the test to
+run the deploy command after the test validator has started.
+
+```typescript
+import { execSync } from "child_process"
+
+...
+
+const deploy = () => {
+  const deployCmd = `solana program deploy --url localhost -v --program-id $(pwd)/target/deploy/config-keypair.json $(pwd)/target/deploy/config.so`
```

Comment:
`deploy_command` see naming in contributing.

Reply:
I would probably also use node (rather than subshells) for `working_directory`, just so we focus on Solana (a lot of people don't understand subshells, hell most people don't understand unix).

Reply:
Yes understood. I have now changed the snippet to the node version for the working directory.

Reply:
Also renamed variable to `deploy_command`.